src/adapter/src/coord/ddl.rs

```diff
@@ -442,6 +443,16 @@ impl Coordinator {
             if !clusters_to_drop.is_empty() {
                 for cluster_id in clusters_to_drop {
                     self.controller.drop_cluster(cluster_id);
+                    for id in self
+                        .catalog()
+                        .get_cluster(cluster_id)
+                        .log_indexes
+                        .values()
+                        .cloned()
+                        .collect_vec()
+                    {
+                        self.drop_compute_read_policy(&id);
+                    }
```

Comment:
Two notes: 
1. I would move this to before we drop the cluster. It's possible that `drop_cluster` wipes out some metadata needed to update the read policies of the log indexes.
2. You'll probably need to collect the log indexes before executing the catalog transaction. Since the transaction completed and this cluster was dropped, the cluster won't exist in the catalog.

Reply:
Thanks!